File: scratch/dung-huong-dan.py
# -*- coding: utf-8 -*-
# Dung src/utils/noiDungHuongDan.ts TU docs/huong-dan-soan-bai.md.
# Mot nguon duy nhat: sua tep .md roi chay lai tep nay (hoac node scratch/dung-huong-dan.mjs,
# hai ban ra cung mot tep).
import io

G = 'D:/claude/math-lms/'
src = io.open(G + 'docs/huong-dan-soan-bai.md', encoding='utf-8').read().replace('\r\n', '\n')

# Bo dong tieu de dau vi hop da co tieu de rieng
body = src.split('\n', 1)[1].lstrip('\n')

# Chuoi template JS: phai chan dau nguoc, dau huyen va ${
esc = body.replace('\\', '\\\\').replace('`', '\\`').replace('${', '\\${')

out = (
    u'/**\n'
    u' * Noi dung trang huong dan soan bai - MOT nguon dung cho ca hai cho:\n'
    u' * hop "Huong dan" trong trinh soan va tep docs/huong-dan-soan-bai.md.\n'
    u' *\n'
    u' * SUA O docs/huong-dan-soan-bai.md roi chay python scratch/dung-huong-dan.py\n'
    u' * (hoac node scratch/dung-huong-dan.mjs - hai ban ra cung mot tep),\n'
    u' * dung sua thang tep nay.\n'
    u' */\n\n'
    u'export const NOI_DUNG_HUONG_DAN = `' + esc + u'`;\n'
)
io.open(G + 'src/utils/noiDungHuongDan.ts', 'w', encoding='utf-8', newline='\n').write(out)
print('xong, %d ky tu' % len(esc))

File: scratch/dung-huong-dan.mjs
// Dựng src/utils/noiDungHuongDan.ts TỪ docs/huong-dan-soan-bai.md (bản Node thay cho
// scratch/dung-huong-dan.py — máy này không có Python). Một nguồn duy nhất: sửa tệp .md rồi chạy:
//   node scratch/dung-huong-dan.mjs
import { readFileSync, writeFileSync } from 'fs';
const src = readFileSync('docs/huong-dan-soan-bai.md', 'utf8').replace(/\r\n/g, '\n');
const body = src.split('\n').slice(1).join('\n').replace(/^\n+/, '');
const esc = body.replace(/\\/g, '\\\\').replace(/`/g, '\\`').replace(/\$\{/g, '\\${');
const out = `/**
 * Noi dung trang huong dan soan bai - MOT nguon dung cho ca hai cho:
 * hop "Huong dan" trong trinh soan va tep docs/huong-dan-soan-bai.md.
 *
 * SUA O docs/huong-dan-soan-bai.md roi chay node scratch/dung-huong-dan.mjs,
 * dung sua thang tep nay.
 */

export const NOI_DUNG_HUONG_DAN = \`${esc}\`;
`;
writeFileSync('src/utils/noiDungHuongDan.ts', out);
console.log('xong,', esc.length, 'kí tự');
